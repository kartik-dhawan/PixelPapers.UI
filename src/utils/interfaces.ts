export interface ChildrenProps {
  children: React.ReactNode
}

export type ContentfulFields = {
  [key: string]: any
}

export interface ThumbnailItem {
  fields: {
    blogMetaTitle: string
    blogTravelLocation: string
    blogType: string
    blogMetaImage: any
  }
  sys: any
}
export interface SocialMediaLink {
  id: number
  label: string
  url: string
  icon: string
}

export interface TravelBlogPost {
  blogTitle?: string
  blogMetaImage?: any
  blogMetaTitle?: string
  blogTravelLocation?: string
  blogType?: string
  blogUrlSlug?: string
  blogInitialDescription?: any
  blogThemeColor?: string
}

export interface ContentfulTravelBlog {
  fields: TravelBlogPost
  sys: any
  metadata: any
}
