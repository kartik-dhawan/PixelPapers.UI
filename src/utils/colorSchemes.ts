export const GLOBAL_COLORS = {
  TEXT_PRIMARY_DARK: "#222",
  TEXT_PRIMARY_LIGHTER: "#666",
  TEXT_PRIMARY_LIGHT: "#d9d9d9",
  BACKGROUND_PRIMARY_LIGHT: "#fefefe",
  BACKGROUND_PRIMARY_DARK: "#111",
  COMMON_FOOTER_BACKGRUND_DARK: "#383b3e",
  BLOG_SUB_COMPONENT_BACKGROUND_LIGHTER: "#efefef",
}

export const PAGE_LEVEL_COLORS = {
  PAGE_404: {
    BACKGROUND_PRIMARY_LIGHT: "#226bb152",
    TEXT_PRIMARY_LIGHT: "#444",
    TEXT_PRIMARY_DARK: "#1c354c",
  },
  TRAVEL: {
    BACKGROUND_PRIMARY_LIGHT: "#fdeded",
    BACKGROUND_PRIMARY_DARK: "#864040",
    TEXT_PRIMARY_LIGHTER: "#444",
  },
}
