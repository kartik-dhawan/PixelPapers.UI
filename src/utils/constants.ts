export const CURATED_BY_LINK_DEFAULT_TEXT = "kartik-dhawan"

export const AVERAGE_TIME_TO_READ_A_BLOG /**In minutes */ = 7
