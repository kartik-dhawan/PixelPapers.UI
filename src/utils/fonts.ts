import { DM_Sans, Anton, Libre_Barcode_39 } from "next/font/google"

export const dmSans = DM_Sans({
  weight: ["200", "300", "400", "500", "600", "700"],
  subsets: ["latin"],
  display: "swap",
  preload: true,
})

export const barcode = Libre_Barcode_39({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})

export const anton = Anton({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})
