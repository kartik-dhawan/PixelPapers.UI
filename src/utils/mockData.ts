export const TRAVEL_PAGE_THUMBNAIL_CARD_DATA = {
  blogTitle: "Top 10 Things to do on your next trip to Jibhi/Jalori Pass.",
  blogMetaImage: {
    metadata: {
      tags: [],
    },
    sys: {
      space: {
        sys: {
          type: "Link",
          linkType: "Space",
          id: "1v98waf66sfj",
        },
      },
      id: "6TVtXnjJIZ812Bxm6VBqz5",
      type: "Asset",
      createdAt: "2023-11-21T17:39:08.757Z",
      updatedAt: "2023-11-21T17:39:17.165Z",
      environment: {
        sys: {
          id: "master",
          type: "Link",
          linkType: "Environment",
        },
      },
      revision: 2,
      locale: "en-US",
    },
    fields: {
      title: "Jibhi_Jalori",
      description:
        "A picture of sunset from Raghupur fort close to jalori pass",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/6TVtXnjJIZ812Bxm6VBqz5/5b89ce3dced278570a2a663453d69a65/IMG_3749.webp",
        details: {
          size: 71536,
          image: {
            width: 3024,
            height: 4032,
          },
        },
        fileName: "IMG_3749.webp",
        contentType: "image/webp",
      },
    },
  },
  blogMetaTitle: "Jibhi - Jalori",
  blogTravelLocation: "Himachal Pradesh",
  blogType: "Travel Itinerary",
}
