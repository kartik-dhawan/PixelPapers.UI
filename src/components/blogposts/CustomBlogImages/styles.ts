import { GLOBAL_COLORS } from "@/utils/colorSchemes"

export const styles = {
  customImage: {
    height: {
      xs: "250px",
      sm: "300px",
      md: "400px",
      xl: "450px",
    },
    width: "100%",
    maxWidth: "900px",
    position: "relative",
    "& img": {
      objectFit: "cover",
    },
  },
  customImageTitleText: {
    textAlign: "center",
    margin: "8px 0px",
    color: GLOBAL_COLORS.TEXT_PRIMARY_LIGHTER,
  },
}
