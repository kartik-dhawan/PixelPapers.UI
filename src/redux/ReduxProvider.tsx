"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
