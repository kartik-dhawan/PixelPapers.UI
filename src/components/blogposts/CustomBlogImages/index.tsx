"use client"

import { Box, Skeleton, Stack } from "@mui/material"
import Image from "next/image"
import { styles } from "./styles"
import { useCallback, useState } from "react"

/** A custom component to render images
 *
 * ===== FORMAT =====
 *
 * ==========================
 *          <IMAGE>
 *      <SUBTITLE TEXT>
 *
 *        <DESCRIPTION>
 * ==========================
 *
 * ==========================
 *          <IMAGE>
 *      <SUBTITLE TEXT>
 *
 *        <DESCRIPTION>
 * ==========================
 */

const CustomBlogImages = ({ blogImages }: { blogImages: any[] }) => {
  const [isImageLoaded, setIsImageLoaded] = useState(false)

  const imageLoadingHandler = useCallback(() => {
    setIsImageLoaded(true)
  }, [])

  return (
    <>
      {blogImages.length !== 0 && (
        <Stack alignItems="center" id="introduction">
          <Box sx={styles.customImage}>
            {!isImageLoaded && (
              <Skeleton variant="rectangular" height="100%" width="100%" />
            )}
            <Image
              src={"https:" + blogImages[0]?.fields.file.url}
              alt={blogImages[0]?.fields.description}
              fill
              sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw)"
              onLoad={imageLoadingHandler}
            />
          </Box>
          <Box sx={styles.customImageTitleText}>
            {blogImages[0]?.fields.title}
          </Box>
          {/* DESCRIPTION will come here */}
        </Stack>
      )}
    </>
  )
}

export default CustomBlogImages
