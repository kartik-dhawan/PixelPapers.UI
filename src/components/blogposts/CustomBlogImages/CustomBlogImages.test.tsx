import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import CustomBlogImages from "."
import { BLOG_IMAGES_ARRAY } from "@/utils/mockData"

describe("Custom Images Component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <CustomBlogImages blogImages={BLOG_IMAGES_ARRAY} />
      </ReduxProvider>,
    )
    expect(tree).toMatchSnapshot()
  })
})
