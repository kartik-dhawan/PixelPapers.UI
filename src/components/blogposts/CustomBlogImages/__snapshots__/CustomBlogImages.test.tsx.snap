// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Images Component -- should match the snapshot 1`] = `
<div
  className="MuiStack-root css-163ljxj-MuiStack-root"
>
  <div
    className="MuiBox-root css-1gi5fv0"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <img
      alt="An image taken on top of Jalori pass' Serolsar Lake."
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw)"
      src="/_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimag%2FIMG_5873.jpg&w=3840&q=75 3840w"
      style={
        {
          "bottom": 0,
          "color": "transparent",
          "height": "100%",
          "left": 0,
          "objectFit": undefined,
          "objectPosition": undefined,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="MuiBox-root css-mr9ztd"
  >
    Serolsar Lake, Jalori pass // Image by the author
  </div>
</div>
`;
